src/core/middlewares/vulnerable/handler.go

```diff
@@ -37,44 +41,100 @@ func New(next http.Handler) http.Handler {
 // ServeHTTP ...
 func (vh vulnerableHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 	imgRaw := req.Context().Value(util.ImageInfoCtxKey)
-	if imgRaw == nil || !config.WithClair() {
+	if imgRaw == nil {
 		vh.next.ServeHTTP(rw, req)
 		return
 	}
-	img, _ := req.Context().Value(util.ImageInfoCtxKey).(util.ImageInfo)
-	if img.Digest == "" {
+
+	// Expected artifact specified?
+	img, ok := imgRaw.(util.ImageInfo)
+	if !ok || len(img.Digest) == 0 {
 		vh.next.ServeHTTP(rw, req)
 		return
 	}
+
+	// Is vulnerable policy set?
 	projectVulnerableEnabled, projectVulnerableSeverity, wl := util.GetPolicyChecker().VulnerablePolicy(img.ProjectName)
 	if !projectVulnerableEnabled {
 		vh.next.ServeHTTP(rw, req)
 		return
 	}
-	vl, err := scan.VulnListByDigest(img.Digest)
+
+	// Invalid project ID
+	if wl.ProjectID == 0 {
+		err := errors.Errorf("project verification error: project %s", img.ProjectName)
+		vh.sendError(err, rw)
+
+		return
+	}
+
+	// Is scanner available?
+	yes, err := scanner.DefaultController.IsScannerAvailable(wl.ProjectID)
```

Comment:
Is it possible the scanner becomes unavailable after scanning?  In that case the policy should still take effect, correct?

Reply:
Of course, we can only do the check based on the current status. I think it's ok for the case you mentioned, there will be corresponding errors returned in the log.

Reply:
But that means, if the scanner is down, the vulnerable images can be pulled even project admin sets the policy?

I don't think it's correct.

Reply:
@reasonerjt 
You should check the code.

I think you misunderstood it. Obviously, the availability is not related to the health of the scanner.  Here I only check the scanner metadata.

I think this check can be removed. It can be depended on the get summary method.